core/src/main/scala/kafka/tools/DumpLogSegments.scala

```diff
@@ -651,4 +651,36 @@ object DumpLogSegments {
     def checkArgs(): Unit = CommandLineUtils.checkRequiredArgs(parser, options, filesOpt)
 
   }
+
+  /*
+   * The kafka.serializer.Decoder is deprecated in 3.8.0. This method is used to transfer the deprecated
+   * decoder to the new org.apache.kafka.tools.api.Decoder. Old decoders have an input VerifiableProperties.
+   * Remove it in new interface since it's always empty.
+   */
+  private[tools] def newDecoder(className: String): Decoder[_] = {
+    try {
+      CoreUtils.createObject[org.apache.kafka.tools.api.Decoder[_]](convertDeprecatedDecoderClass(className))
+    } catch  {
+      case _: Exception =>
+        val decoder = CoreUtils.createObject[kafka.serializer.Decoder[_]](className, new VerifiableProperties())
```

Comment:
Please add comments to say "we always pass empty VerifiableProperties since this tool can't take custom configs ... bababa"